packages/core/src/routes/role.ts

```diff
@@ -180,4 +186,72 @@ export default function roleRoutes<T extends AuthedRouter>(router: T) {
       return next();
     }
   );
+
+  router.get(
+    '/roles/:id/users',
+    koaGuard({
+      params: object({ id: string().min(1) }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+
+      await findRoleById(id);
+      const usersRoles = await findUsersRolesByRoleId(id);
+      ctx.body = await findUsersByIds(usersRoles.map(({ userId }) => userId));
+
+      return next();
+    }
+  );
+
+  router.post(
+    '/roles/:id/users',
+    koaGuard({
+      params: object({ id: string().min(1) }),
+      body: object({ userIds: string().min(1).array() }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+        body: { userIds },
+      } = ctx.guard;
+
+      await findRoleById(id);
+      const usersRoles = await findUsersRolesByRoleId(id);
+
+      for (const userId of userIds) {
+        assertThat(
+          !usersRoles.some(({ userId: _userId }) => _userId === userId),
+          new RequestError({
```

Comment:
Can we skip this check?
Since this API assigns the role to many users, It seems ok if one of the users already has this role.

Reply:
Is not OK? At least for current design, this won't happen

Reply:
I assume yijun means whether we should throw the error out and interrupt this request or silently return success.

Reply:
If we want to allow "skip", then better confirm with PM